src/front/domain_socket.rs

```diff
@@ -248,3 +248,22 @@ impl DomainSocketListenerBuilder {
         })?)
     }
 }
+
+impl GetMetadata for UnixStream {
+    #[cfg(feature = "peer-credentials-authenticator")]
+    fn metadata(&self) -> Option<ConnectionMetadata> {
+        let ucred = self
+            .peer_cred()
+            .expect("Failed to get peer credentials for Unix domain socket connection.");
```

Comment:
I would prefer returning an error instead of panicking. In our case, you can log something and return `None`!